pdata/internal/generated_pmetric.go

```diff
@@ -1489,8 +1489,20 @@ func (ms HistogramDataPoint) CopyTo(dest HistogramDataPoint) {
 		dest.SetSum(ms.Sum())
 	}
 
-	dest.SetBucketCounts(ms.BucketCounts())
-	dest.SetExplicitBounds(ms.ExplicitBounds())
+	if ms.orig.BucketCounts != nil {
```

Comment:
```golang
if len(ms.orig.BucketCounts) == 0 {
	dest.orig.BucketCounts = nil
} else {
	dest.orig.BucketCounts = make([]uint64, len(ms.orig.BucketCounts))
	copy(dest.orig.BucketCounts, ms.orig.BucketCounts)
}
```

Reply:
But this would be not exact copy... if `ms.orig.BucketCounts` has `[]uint64{}`, `dest.orig.BucketCounts` will be `nil`

Reply:
and? do we need to distinguished between these?